### ISSUE TITLE
[Android] Push Notification cannot popping up when received in the background

### ISSUE THREAD

**Issue by richso** [zo0r/react-native-push-notification#2115]
I revise my App from React Native 0.5x to 0.64 and react-native-push-notification form 3.1.3 to 7.4.0.

My problem is that the, in new version, the Push Notification received while the App is in background or killed cannot be popping up and no sound and vibration either BUT no problem for foreground and totally OK for IOS; checked  with the Android Studio "console.log()" dump, the App should have received the message in the background. 

The PushNotification.configure JSON is the same for both old and new version, but the old version works for background cases. Am I missing something?

file: AndroidManifest.xml
`<manifest xmlns:android="http://schemas.android.com/apk/res/android"
  package="hk.kinliu.app">

    <uses-permission android:name="android.permission.INTERNET" />
    <uses-permission android:name="android.permission.VIBRATE" />
    <uses-permission android:name="android.permission.RECEIVE_BOOT_COMPLETED"/>
    <uses-permission android:name="android.permission.USE_FULL_SCREEN_INTENT"/>

    <application
      android:name=".MainApplication"
      android:label="@string/app_name"
      android:icon="@mipmap/ic_launcher"
      android:roundIcon="@mipmap/ic_launcher_round"
      android:allowBackup="false"
      android:theme="@style/AppTheme">
      <activity
        android:name=".MainActivity"
        android:label="@string/app_name"
        android:configChanges="keyboard|keyboardHidden|orientation|screenSize|uiMode"
        android:launchMode="singleTask"
        android:windowSoftInputMode="adjustResize">
        <intent-filter>
            <action android:name="android.intent.action.MAIN" />
            <category android:name="android.intent.category.LAUNCHER" />
        </intent-filter>
      </activity>

        
        <meta-data  android:name="com.dieam.reactnativepushnotification.notification_foreground"
            android:value="false"/>
        
        <meta-data  android:name="com.dieam.reactnativepushnotification.notification_color"
            android:resource="@android:color/white"/>

        <receiver android:name="com.dieam.reactnativepushnotification.modules.RNPushNotificationActions" />
        <receiver android:name="com.dieam.reactnativepushnotification.modules.RNPushNotificationPublisher" />
        <receiver android:name="com.dieam.reactnativepushnotification.modules.RNPushNotificationBootEventReceiver">
            <intent-filter>
                <action android:name="android.intent.action.BOOT_COMPLETED" />
                <action android:name="android.intent.action.QUICKBOOT_POWERON" />
                <action android:name="com.htc.intent.action.QUICKBOOT_POWERON"/>
            </intent-filter>
        </receiver>

        <service
            android:name="com.dieam.reactnativepushnotification.modules.RNPushNotificationListenerService"
            android:exported="false" >
            <intent-filter>
                <action android:name="com.google.firebase.MESSAGING_EVENT" />
            </intent-filter>
        </service>
    </application>
</manifest>
`

file: App.js (Excerpt)
`async componentWillMount() {
    	var self = this;
      PushNotification.configure({

            ......

            // (optional) Called when Token is generated (iOS and Android)
            onRegister: function(token) {
                console.log( '@@@###TOKEN:', token );
                
                var res = ajax.registerPushToken({
                    os: Platform.OS == 'ios' ? 'iOS' : 'Android',
                    token: token.token
                });
                console.log('@@@###sync TOKEN:', res);
                
                self.setState({
                    push_token: token
                });
            },

            onNotification: async function(data) {
            	console.log("@@@###note data", JSON.stringify(data));
                
                if (data.foreground) {
                    Alert.alert(
                        "Recommended\n" + data.message, 
                        "go to the article?",
                        [{
                            text: 'OK',
                            onPress:() => self.gotoArticle(data.data.id)
                        },{
                            text: 'Cancel',
                            style: 'cancel'
                        }],
                        {
                            cancelable: true
                        }
                    );
                } else if (data.userInteraction) {
                	     ....
                }
                
                PushNotification.setApplicationIconBadgeNumber(0);
                data.finish(PushNotificationIOS.FetchResult.NoData); 
            },

            senderID: 'xxxxxxxxxxxxxxxxxxx',

            permissions: {
                alert: true,
                badge: true,
                sound: true
            },

            popInitialNotification: true,

            requestPermissions: true,
        });`

**Comment by Dallas62** [comment]
Hi @richso,
Please refer to CHANGELOG when upgrading.
Every changes are documented.
Regards

**Comment by Rushiraj-karm** [comment]
Hi @richso 
is your problem solved ?
can you please share solution
